### ISSUE TITLE
SC2000 check inconsistent

### ISSUE THREAD

**Issue by cslycord** [koalaman/shellcheck#1938]
#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```bash

#!/bin/bash
echo "$1" | wc -c
echo -n "$1" | wc -c
printf "%s" "$1" | wc -c

```

#### Here's what shellcheck currently says:
when using plain echo (without any arguments), shellcheck warns to see if you can replace it with ${#variable}

When "echo -n" or "printf" are used that ensure no newline is in the output stream, no warning is issued

#### Here's what I wanted or expected to see:
The warning to be used in all 3 cases
